### PR TITLE
feat(aws-docdb-cluster): enable the ssm parameter store to record the…

### PR DESCRIPTION
**Pull request by haidargit**
## what

In this PR, we can use the Cloud Posse [ssm parameter store module](https://github.com/cloudposse/terraform-aws-ssm-parameter-store) to store the documentdb _master_password_ information.

## why

The objective behind this PR is to ensure the secure distribution of the docdb cluster's master password within the AWS infrastructure. We can centrally manage and protect sensitive information, increasing operational efficiency.

## references

_No issue relates to the current improvement._

I have run these required commands.
`make init`
`make readme`

Kindly review this PR for documentdb module improvements. Thank you, Cloud Posse Team!

### CODE REVIEW

**Comment by kevcube** [comment]
@haidargit i think you should use [this module](https://github.com/cloudposse/terraform-aws-ssm-parameter-store) instead of the resource

File: variables.tf [hunk 1]
```diff
@@ -204,3 +204,21 @@ variable "ca_cert_identifier" {
...
+  type        = bool
+  default     = false
+  description = "Whether an SSM parameter store value is created to store the key's private key pem."
+}
+
+variable "ssm_parameter_type" {
+  type        = string
+  default     = "SecureString"
+  description = "Type of the ssm parameter. Valid types are String, StringList and SecureString."
+}
```
Comment (kevcube) [inline variables.tf:218]: Probably doesn't need to be a variable, it should not be allowed to store as `String`

File: variables.tf [hunk 2]
```diff
@@ -204,3 +204,21 @@ variable "ca_cert_identifier" {
   description = "The identifier of the CA certificate for the DB instance"
   default     = null
 }
+
+variable "ssm_parameter_enabled" {
+  type        = bool
+  default     = false
+  description = "Whether an SSM parameter store value is created to store the key's private key pem."
```
Comment (kevcube) [inline variables.tf:211]:
```suggestion
  description = "Whether an SSM parameter store value is created to store the database password."
```

**Review by kevcube — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 1 above)
Comment (haidargit) [reply variables.tf:218]: okay, will store it as `SecureString` for the default on this resource. thank you @kevcube

File: variables.tf (hunk 2 above)
Comment (haidargit) [reply variables.tf:211]: sorry @kevcube, i forgot to edit this one. i will fix it

**Comment by haidargit** [comment]
Hi @kevcube, i have updated the PR to add the changes, kindly review the code. Thank you

File: variables.tf [hunk 3]
```diff
@@ -204,3 +204,21 @@ variable "ca_cert_identifier" {
...
+  type        = string
+  default     = "/docdb/master-password/"
+  description = "The path prefix for the created SSM parameter e.g. '/docdb/master-password/dev'. `ssm_parameter_enabled` must be set to `true` for this to take affect."
+}
+
+variable "ssm_parameter_description" {
+  type        = string
+  default     = "documentdb cluster master_password"
+  description = "Description of the parameter"
+}
```
Comment (kevcube) [inline variables.tf:224]:
```suggestion
```

File: main.tf [hunk 4]
```diff
@@ -152,3 +152,20 @@ module "dns_replicas" {
...
+  source  = "cloudposse/ssm-parameter-store/aws"
+  version = "0.11.0"
+
+  enabled = module.this.enabled && var.ssm_parameter_enabled == true ? true : false 
+  parameter_write = [
+    {
+      name        = format("%s%s", var.ssm_parameter_path_prefix, module.this.id)
+      value       = var.master_password != "" ? var.master_password : random_password.password[0].result
+      type        = "String"
+      description = var.ssm_parameter_description
```
Comment (kevcube) [inline main.tf:166]: Master password for `module.this.id` DocumentDB cluster

File: variables.tf (hunk 3 above)
Comment (kevcube) [reply variables.tf:224]: too generic. let's just hardcode description based on cluster's name.

**Review by kevcube — CHANGES_REQUESTED** [review]
please enable the ssm param in `examples/complete` so it gets tested

**Comment by haidargit** [comment]
Could you help review again, @kevcube ? i have added the changes. Thanks

**Comment by kevcube** [comment]
@haidargit see failing checks.

Delete your `.terraform.lock.hcl` file before running `make readme` and it should properly generate the README.

**Comment by haidargit** [comment]
Thanks @kevcube! i have removed the lock files and updated the readme. 
could you help check again?

**Review by kevcube — APPROVED** [review]
thanks for the contribution!

**Comment by haidargit** [comment]
Thanks Cloud Posse, @kevcube !!

**Pull request merged**